ftl/code/controllers/subsystem/ftl_navigation.dm

```diff
@@ -0,0 +1,198 @@
+SUBSYSTEM_DEF(ftl_navigation)
+	name = "FTL Navigation"
+	wait = 10
+	init_order = INIT_ORDER_FTL_NAV
+	flags = SS_NO_FIRE
+
+	var/ftl_state = FTL_IDLE //is the ship in transit, spooling or idle?
+
+	var/datum/sector/current_sector
+	var/datum/sector/to_sector
+
+	var/list/visited_sectors = list()
+
+	var/sector_count = 0
+
+	var/list/ship_areas = list()
+
+	var/list/sector_name_fragments = list("le", "xe", "ge", "za", "ce", "bi", "so", "us", "es", "ar", "ma", "in", "di", "re", "a", "er", "at", "en", "be", "ra", "la", "ve", "ti", "ed", "or", "gu", "an", "te", "is", "ri", "on")
+
+	var/list/sector_types = list()
+	var/list/planet_types = list()
+
+	var/jump_knockdown_force = 3
+	var/jump_throw_force = 0 //Badmin var
+
+
+/datum/controller/subsystem/ftl_navigation/Initialize(timeofday)
+	for(var/A in get_areas(/area/, TRUE)) //Get all areas owned by the ship
+		var/area/place = A
+		if(place.z != 2) continue
+		ship_areas += place
+
+	for(var/type in subtypesof(/datum/planet)) //Get all planet types and weights
+		var/datum/planet/P = type
+		planet_types += P
+		planet_types[type] = initial(P.frequency)
+
+	for(var/type in subtypesof(/datum/sector)) //Get all sector types and weights
+		var/datum/sector/S = type
+		if(initial(S.frequency) > 0)
+			sector_types += S
+			sector_types[type] = initial(S.frequency)
+
+
+	current_sector = new /datum/sector/named/nt_home //Only give them a basic sector to start in
+	visited_sectors += current_sector
+	generate_connecting_sectors(current_sector)
+	for(var/s in current_sector.connected_sectors)
+		var/datum/sector/S = s
+		generate_connecting_sectors(S)
+
+	var/timer = world.timeofday
+	var/count = 0
+	var/msg
+	for(var/z in SSmapping.levels_by_trait(ZTRAIT_STATION))
+		for(var/datum/sub_turf_block/STB in split_block(locate(1,1,z), locate(255,255,z)))

```

Comment:
What's the point of this sub_turf_block?

Reply:
This STB loops through the ship's z level replacing normal space turfs with our custom transit ones.

If it detects any it sends a message during init that it found X tiles and took Y seconds to fix, like how the atmos subsystem reports any roundstart active atmos turfs

Reply:
No that's what the loop is doing, why is this using sub turf blocks instead of a simple
`for(var/turf/place in block(locate(1,1,z), locate(255,255,z)))`